crates/uv/src/commands/tool/install.rs

```diff
@@ -0,0 +1,135 @@
+use std::fmt::Write;
+use std::str::FromStr;
+
+use anyhow::{bail, Context, Result};
+use distribution_types::Name;
+use pep508_rs::Requirement;
+
+use pypi_types::VerbatimParsedUrl;
+use tracing::debug;
+use uv_cache::Cache;
+use uv_client::Connectivity;
+use uv_configuration::{Concurrency, PreviewMode};
+use uv_fs::{replace_symlink, Simplified};
+use uv_installer::SitePackages;
+use uv_requirements::RequirementsSource;
+use uv_tool::{entrypoint_paths, find_executable_directory, InstalledTools, Tool};
+use uv_toolchain::{EnvironmentPreference, Toolchain, ToolchainPreference, ToolchainRequest};
+use uv_warnings::warn_user;
+
+use crate::commands::project::update_environment;
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+use crate::settings::ResolverInstallerSettings;
+
+/// Install a tool.
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn install(
+    name: String,
+    python: Option<String>,
+    from: Option<String>,
+    with: Vec<String>,
+    settings: ResolverInstallerSettings,
+    _isolated: bool,
```

Comment:
What's with this argument?

Reply:
Copied over from `tool/run.rs` — I'm not sure.